src/system_structure.jl

```diff
@@ -0,0 +1,851 @@
+# Copyright (c) 2025 Bart van de Lint
+# SPDX-License-Identifier: MPL-2.0
+
+"""
+    const SimFloat = Float64
+
+This type is used for all real variables used in simulations.
+"""
+const SimFloat = Float64
+
+"""
+   const KVec3 = MVector{3, SimFloat}
+
+Basic 3-dimensional vector, stack allocated, mutable.
+"""
+const KVec3 = MVector{3, SimFloat}
+
+"""
+   const KVec4 = MVector{4, SimFloat}
+
+Basic 4-dimensional vector, stack allocated, mutable.
+"""
+const KVec4 = MVector{4, SimFloat}
+
+"""
+   const SVec3 = SVector{3, SimFloat}
+
+Basic 3-dimensional vector, stack allocated, immutable.
+"""
+const SVec3 = SVector{3, SimFloat}
+
+"""
+    SegmentType `POWER_LINE` `STEERING_LINE` `BRIDLE`
+
+Enumeration for the type of a tether segment.
+
+# Elements
+- `POWER_LINE`: A segment belonging to a main power line.
+- `STEERING_LINE`: A segment belonging to a steering line.
+- `BRIDLE`: A segment belonging to the bridle system.
+"""
+@enum SegmentType begin
+    POWER_LINE
+    STEERING_LINE
+    BRIDLE
+end
+
+"""
+    DynamicsType `DYNAMIC` `QUASI_STATIC` `WING` `STATIC`
+
+Enumeration for the dynamic model governing a point's motion.
+
+# Elements
+- `DYNAMIC`: The point is a dynamic point mass, moving according to Newton's second law.
+- `QUASI_STATIC`: The point's acceleration is constrained to zero, representing a force equilibrium.
+- `WING`: The point is rigidly attached to a wing body and moves with it.
+- `STATIC`: The point's position is fixed in the world frame.
+"""
+@enum DynamicsType begin
+    DYNAMIC
+    QUASI_STATIC
+    WING
+    STATIC
+end
+
+"""
+    mutable struct Point
+
+A point mass, representing a node in the mass-spring system.
+
+$(TYPEDFIELDS)
+"""
+mutable struct Point
+    const idx::Int16
+    const transform_idx::Int16 # idx of wing used for initial orientation
+    const wing_idx::Int16
+    const pos_cad::KVec3
+    const pos_b::KVec3 # pos relative to wing COM in body frame
+    const pos_w::KVec3 # pos in world frame
+    const vel_w::KVec3 # vel in world frame
+    const disturb::KVec3 # disturbing force
+    const force::KVec3
+    const type::DynamicsType
+    mass::SimFloat
```

Comment:
Documentation of the fields mass and bridle_damping is missing. A physical unit must be provided. How can a point have a bridle_damping? Damping is a property of a segment, and not of a point.